### PR TITLE
Add a wrapper for prove, verify and param derivation.

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Content

A simple wrapper struct around `prove()` and `verify()` functions and parameter derivation.

## Pre-submit checklist

- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
- PR
    - [x] No clippy warnings in the CI
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- Documentation
    - [x] Update README file (if relevant)
    - [x] Update documentation website (if relevant)

## Issue(s)

Relates to #66

### CODE REVIEW

File: src/centralized_telescope/wrapper.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use super::setup::Setup;
+use crate::utils::types::Element;
+
+/// The main ALBA struct with prove and verify functions.
+#[derive(Debug, Clone, Copy)]
+pub struct Wrapper {
+    setup: Setup,
+}
+
+impl Wrapper {
```
Comment (rrtoledo) [inline src/centralized_telescope/wrapper.rs:14]: Why not calling this CentralizedTelescope directly?

File: src/centralized_telescope/wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /// Initialize ALBA with `Params`.
```
Comment (rrtoledo) [inline src/centralized_telescope/wrapper.rs:15]: Can you add doc strings for all public functions?

File: src/centralized_telescope/wrapper.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    pub fn create(params: &Params) -> Self {
+        let setup = make_setup(params);
+        Self::create_with_setup(&setup)
+    }
+
+    /// Initialize ALBA with `Setup`. This function is unsafe to use and should
```
Comment (rrtoledo) [inline src/centralized_telescope/wrapper.rs:21]: I would put the warning first

File: src/centralized_telescope/wrapper.rs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    /// be avoided.
+    pub fn create_with_setup(setup: &Setup) -> Self {
```
Comment (rrtoledo) [inline src/centralized_telescope/wrapper.rs:23]: I would add a `_unsafe` at the end of the name

File: src/centralized_telescope/wrapper.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    pub fn prove(&self, prover_set: &[Element]) -> Option<Proof> {
+        algorithm::prove(&self.setup, prover_set)
+    }
+
+    /// Alba's verification algorithm.
+    /// Returns true if and only if the proof is successfully verified.
+    pub fn verify(&self, proof: &Proof) -> bool {
+        algorithm::verify(&self.setup, proof)
+    }
+}
```
Comment (rrtoledo) [inline src/centralized_telescope/wrapper.rs:38]: The tests should be moved to the user facing wrapper

**Review by rrtoledo — CHANGES_REQUESTED** [review]

File: src/centralized_telescope/wrapper.rs (hunk 1 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs:14]: I was thinking along the lines of: inside the `centralized_telescope` directory, `CentralizedTelescope` doesn't say much, so I called it `Wrapper` and put it in `wrapper.rs`. For users outside of `centralized_telescope`, however, it is re-exported as  `CentralizedTelescope`. What do you think?

File: src/centralized_telescope/wrapper.rs (hunk 3 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs]: done

File: src/centralized_telescope/wrapper.rs (hunk 4 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs:23]: Preference between `create_with_setup_unsafe` and `create_unsafe`?

File: src/centralized_telescope/wrapper.rs (hunk 5 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs:38]: I agree that tests should be moved. Let's do this in a separate PR?

File: src/centralized_telescope/wrapper.rs (hunk 4 above)
Comment (rrtoledo) [reply src/centralized_telescope/wrapper.rs:23]: Either is fine

File: src/centralized_telescope/wrapper.rs (hunk 2 above)
Comment (rrtoledo) [reply src/centralized_telescope/wrapper.rs:15]: Create an issue and add it to the relevant meta issue then :)

File: src/centralized_telescope/wrapper.rs (hunk 1 above)
Comment (rrtoledo) [reply src/centralized_telescope/wrapper.rs:14]: Sure

File: src/centralized_telescope/wrapper.rs (hunk 5 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs:38]: #79

File: src/centralized_telescope/wrapper.rs (hunk 4 above)
Comment (tolikzinovyev) [reply src/centralized_telescope/wrapper.rs]: done

**Review by rrtoledo — APPROVED** [review]
lgtm

**Review by curiecrypt — APPROVED** [review]

**Pull request merged**